### PR TITLE
Start

### PR DESCRIPTION
**Pull request by Spectral-Kaburu**

### CODE REVIEW

File: algorithms/depth_first.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+class BinaryTree():
+    def __init__(self, data):
+        self.data = data
+        self.left = None
+        self.right = None
+
+    def child(self, data):
```
Comment (MartinKaburu) [inline algorithms/depth_first.py]:
```suggestion
    def insert(self, data):
```
Name functions by their purpose

File: algorithms/depth_first.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        if self.data == data:
+            return
```
Comment (MartinKaburu) [inline algorithms/depth_first.py:9]: You can handle duplicates better, allow them and do so

**Review by MartinKaburu — CHANGES_REQUESTED** [review]
Beautiful stuff. Add Types definitions for all this. it's 2024 and we are not savages

File: algorithms/depth_first.py (hunk 2 above)
Comment (Spectral-Kaburu) [reply algorithms/depth_first.py:9]: I should let duplicates exist???

**Pull request merged**